### ISSUE TITLE
[Models] [Pipelines] Patch Diffusion

### ISSUE THREAD

**Issue by clarencechen** [huggingface/diffusers#2522]
### Model/Pipeline/Scheduler description

The core idea of the paper is to insert a ViT-style patching operation at the beginning of the U-Net, letting it operate on data with smaller height and width. We show in our paper that the optimal prediction for x is quite blurry for most timesteps, and therefore convolutions at the original resolution are usually not necessary. This causes a considerable reduction in compute cost: For example, when using a patch size of 4 (P = 4), generating 256x256 images costs only as much as generating 64x64 images normally (with P = 1).

Pretrained weights are available on Google Drive for ImageNet 256x256 and FFHQ 1024x1024, as well as 3 LSUN models with P=2, P=4, and P=8.

My current design plan for implementing the main contributions from this paper:
- Add a keyword argument to use patching layers in the UNet*Model and Transformer*Model classes.
- Add an inference pipeline that supports model splitting by sample timestep.

### Open source status

- [X] The model implementation is available
- [X] The model weights are available (Only relevant if addition is not a scheduler).

### Provide useful links for the implementation

Author of Original Implementation: Eric Luhman @ericl122333
Paper Abstract: https://arxiv.org/abs/2207.04316
Model Implementation: https://github.com/ericl122333/PatchDiffusion-Pytorch
Links to the Google Drive URLs for the pretrained weights are in the model implementation README.

**Comment by clarencechen** [comment]
Just published a PR to add patching for PyTorch UNet models at #2698

**Comment by a-r-r-o-w** [comment]
Marking as closed due to long term of no activity
